lib/transaction.js

```diff
@@ -180,6 +188,12 @@ Transaction.prototype.prepareEnvironment = function() {
     return self.setIsolationLevel();
   }).then(function () {
     return self.setAutocommit();
+  }).catch(function (err) {
+    try {
+      self.rollback();
+    } catch (e) {}
+
+    throw err;
```

Comment:
how about this? reporting error about setup is more important so we are going to ignore rollback error anyway... what's important is that we should "release" the connection which we will do regardless of any error during rollback!

```javascript
  }).catch(function (setupErr) {
    return self.rollback()
      .finally(function () {
        throw setupErr;
      });
  }).tap(function () {
```